Kernel/FileSystem/Ext2FileSystem.cpp

```diff
@@ -64,13 +64,23 @@ static u8 to_ext2_file_type(mode_t mode)
     return EXT2_FT_UNKNOWN;
 }
 
-NonnullRefPtr<Ext2FS> Ext2FS::create(BlockDevice& device)
+NonnullRefPtr<Ext2FS> Ext2FS::create(File& file)
 {
-    return adopt(*new Ext2FS(device));
+    return adopt(*new Ext2FS(file));
 }
 
-Ext2FS::Ext2FS(BlockDevice& device)
-    : DiskBackedFS(device)
+NonnullRefPtr<Ext2FS> Ext2FS::create(NonnullRefPtr<FileDescription> file_description)
+{
+    return adopt(*new Ext2FS(file_description));
+}
+
+Ext2FS::Ext2FS(File& file)
+    : FileBackedFS(file)
+{
+}
+
+Ext2FS::Ext2FS(NonnullRefPtr<FileDescription> file_description)
+    : FileBackedFS(file_description)
```

Comment:
Again, let's just decide whether we want to take a file or a file description, and always accept that, and make the callers (are there many? :slightly_smiling_face:) get the right one themselves. Also, no need to pass `NonnullRefPtr<FileDescription>` when you can just pass a `FileDescription&`